src/WasmWebTerm.js

```diff
@@ -819,9 +826,17 @@ class WasmWebTerm {
           if (data == "\x08") buffer = buffer.slice(0, -1)
           else buffer += data
 
-          // echo input back (special handling for return, backspace, and escape sequences)
-          if (data == "\n") this._xterm.write("\r\n")
-          else if (data == "\x08") this._xterm.write("^H")
+          // line complete -> return the input buffer
+          if (data == "\n") {
+            // only echo the linebreak when there is no prompt (i.e. we assume multi-line input)
+            if (!message) this._xterm.write("\r\n")
+
+            handler.dispose()
+            return resolve(buffer)
+          }
+
+          // echo input back (special handling for backspace and escape sequences)
+          if (data == "\x08") this._xterm.write("^H")
           else if (data.charCodeAt(0) == 0x1b)
             this._xterm.write("^[" + data.slice(1))
           else this._xterm.write(data)
```

Comment:
Neither wasmer (with WASI) nor emscripten appear to support the POSIX `termios.h` interface to enable/disable local echo. Therefore we don't know if the program wants to disable echoing.